src/lib.rs

```diff
@@ -440,4 +440,78 @@ impl TxBuilder {
     }
 }
 
+struct BumpFeeTxBuilder {
+    txid: String,
+    fee_rate: f32,
+    rbf: Option<RbfValue>,
+    allow_shrinking: Option<String>,
+}
+
+impl BumpFeeTxBuilder {
+    fn new(txid: String, fee_rate: f32) -> Self {
```

Comment:
The `new` associated function returns `Self` whereas the other 3 methods return `Arc<Self>`. Could that create a problem somewhere? Or can the `build` method consume both `self` and `Arc<Self>`?

Reply:
My understanding is that since `Arc<T>`  implements `AsRef<T>` that is can be used as a &T, or &self in this case.  The reason for returning an `Arc<T>` on a function that returns Self is specific to how uniffi works, https://mozilla.github.io/uniffi-rs/udl/interfaces.html#managing-shared-references. And not needing to return Arc<Self> for the constructor we also don't do because uniffi doesn't require it there. There could be a more efficient way to do this were we don't allocate a whole new builder on each call, but given the short life of these structs anyway I don't think it's a problem.